base/libc.jl

```diff
@@ -176,13 +176,20 @@ struct TimeVal
    usec::Int64
 end
 
-function TimeVal()
-    tv = Ref{TimeVal}()
-    status = ccall(:jl_gettimeofday, Cint, (Ref{TimeVal},), tv)
+struct TimeSpec
+   sec::Int64
+   nsec::Int64
+end
+
+function TimeSpec()
+    ts = Ref{TimeSpec}()
+    status = ccall(:jl_gettimeofday, Cint, (Ref{TimeSpec},), ts)
```

Comment:
This could now go through libuv to avoid our need for a custom wrapper (and get 100-nanosecond precision)
```suggestion
    UV_CLOCK_REALTIME = 1
    status = ccall(:uv_clock_gettime, Cint, (Cint, Ref{TimeSpec},), UV_CLOCK_REALTIME, ts)
```
https://docs.libuv.org/en/v1.x/misc.html#c.uv_clock_gettime

Reply:
Does this have resolution guarantees? I can't seem to find that in the libuv docs :thinking: